py/server/deephaven/stream/kafka/producer.py

```diff
@@ -36,11 +37,15 @@ def j_object(self) -> jpy.JType:
 def produce(
     table: Table,
     kafka_config: Dict,
-    topic: str,
+    topic: Optional[str],
     key_spec: KeyValueSpec,
     value_spec: KeyValueSpec,
     last_by_key_columns: bool = False,
     publish_initial: bool = True,
+    partition: Optional[int] = None,
+    topic_column: Optional[str] = None,
+    partition_column: Optional[str] = None,
+    timestamp_column: Optional[str] = None,
 ) -> Callable[[], None]:
     """Produce to Kafka from a Deephaven table.
```

Comment:
I would go over all of the new pydocs to make sure they answer all questions a user may have.  They seem light on details for a number of cases, and I may not know enough cases to fully say where they are not comprehensive enough.